### DISCUSSION TITLE
Moving items in simple boxes and items app

### DISCUSSION THREAD

**Discussion by lekv** [Budibase/budibase#12206] [Help]
I'm trying to build a simple boxes and items app. Boxes have a name and location (kitchen, garage, etc), items have a name, description, picture, and a box. The relationship between the two is 1:n (1 box containing many items). Each item can only be in one box at a time and there are no multiple instances of an item.

So far this all works as expected. However, I want to be able to move items between boxes *from the boxes view*. This is useful when I have an empty box and a bunch of items flying around. Then I just want to put them all into this box and have them removed from whatever box they had been taken from originally.

However, if I pick an item (which is still assigned to its old box) to place it into a new box, I get an error saying *1:N Relationship Error: Record already linked to another.*. I understand why this happens, but I would like to ask if there's a way to change this behavior and instead of failing have the app remove the other, conflicting link and create the new one.

Thanks for the help!

**Comment by ConorWebb96** [comment]
Hey @lekv,

I'm assuming you are using some form of external database as the Budibase internal db doesn't do 1 to 1 relationships. This sounds like you will need to change the type of relationship you are using for the items within the boxes. 

Or you will have to create functionality were you first unlink then assign the item to a new box? This could be something as simple as a multi-step form where the initial step unlinks the item from the box, then second step then allows you to re-link that item to a different box.

I hope this makes sense.

**Reply by lekv** [reply]
Thanks for the response. I'm using the internal Budibase DB with a 1:n relationship (1 box containing many items).

In the meantime I tried using n:m relations and an automation to remove the item from the older box. With the n:m relation it can be in two boxes, the one that it was originally in and the new one I'm adding it to. Then I was hoping the new box would be appended to the list of boxes at the end and I can just truncate it to the last item but that doesn't seem to work — the boxes are in order of their ID.

I might look into a form next. It would be really nice to be able to just quickly add items to a new box and have them automatically removed from their old boxes.
